vllm/v1/attention/backends/mla/flashmla.py

```diff
@@ -184,8 +184,23 @@ def _forward_decode(
             q = torch.cat(q, dim=-1)
 
         assert isinstance(q, torch.Tensor)
+
+        batch_size = attn_metadata.decode.seq_lens.shape[0]
```

Comment:
Could you refactor this into a utility function? It will likely need to be called in each backend that supports this feature (FlashInfer-MLA at least), so it will be nice to be able to reuse the logic.